From 2207bb5e0e1592a28f543e1816aa257988b6c037 Mon Sep 17 00:00:00 2001
From: Karol Niczyj <karol.niczyj@intel.com>
Date: Wed, 14 Jul 2021 15:33:25 +0200
Subject: [PATCH] Added PmtService to Service Config Manager

Added Pmt Service and encoding for '.' character.

Tested manually by checking if pmt service get masked.
---
 src/main.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/main.cpp b/src/main.cpp
index 01cedbf..5af6d9d 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -41,7 +41,7 @@ static std::unordered_map<std::string /* unitName */,
     managedServices = {{"phosphor-ipmi-net", false}, {"bmcweb", false},
                        {"phosphor-ipmi-kcs", false}, {"start-ipkvm", false},
                        {"obmc-console", false},      {"dropbear", true},
-                       {"obmc-console-ssh", true}};
+                       {"obmc-console-ssh", true},   {"xyz.openbmc_project.Pmt", false}};
 
 enum class UnitType
 {
@@ -131,7 +131,9 @@ static inline void
             }
 
             std::string instantiatedUnitName =
-                unitName + addInstanceName(instanceName, "@");
+                unitName + addInstanceName(instanceName, "_40");
+            boost::replace_all(instantiatedUnitName, "-", "_2d");
+            boost::replace_all(instantiatedUnitName, ".", "_2e");
             const sdbusplus::message::object_path& objectPath =
                 std::get<static_cast<int>(ListUnitElements::objectPath)>(unit);
             // Group the service & socket units togther.. Same services
-- 
2.25.1

